staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -245,7 +245,7 @@ func (p *jsonPatcher) originalStrategicMergePatch() (map[string]interface{}, err
 	return originalPatchMap, nil
 }
 
-func (p *jsonPatcher) computeStrategicMergePatch(currentObject runtime.Object) (map[string]interface{}, error) {
+func (p *jsonPatcher) computeStrategicMergePatch(currentObject runtime.Object, _ map[string]interface{}) (map[string]interface{}, error) {
```

Comment:
My understanding is that it still works, you just don't get any of the fancy directives? Which is fine in this case since it will always be applied to the version it is computed from.

Regardless, this function makes the thing the code is already doing more obvious.

Reply:
>> not all types support SMP
>
> My understanding is that it still works, you just don't get any of the fancy directives?

no, it fails to extract schema information and fails SMP computation entirely.

> Regardless, this function makes the thing the code is already doing more obvious.

sure, though it pins the json patcher to an interface we know it can't satisfy. I anticipate the "subsequent retry" behavior will likely go away entirely pending outcome of follow-up discussion in https://github.com/kubernetes/kubernetes/pull/62868#discussion_r183859190 so I don't feel that strongly

Reply:
Adding a TODO.